clients/src/main/java/org/apache/kafka/clients/producer/internals/RecordAccumulator.java

```diff
@@ -264,6 +264,25 @@ public void reenqueue(RecordBatch batch, long now) {
      * </ol>
      */
     public ReadyCheckResult ready(Cluster cluster, long nowMs) {
+        return ready(cluster, null, nowMs);
+    }
+
+    /**
+     * Get a list of nodes whose partitions are ready to be sent, and the earliest time at which any non-sendable
+     * partition will be ready; Also return the flag for whether there are any unknown leaders for the accumulated
+     * partition batches.
+     * <p>
+     * A destination node is ready to send data if ANY one of its partition is not backing off the send and has no
+     * in flight batches and ANY of the following are true :
+     * <ol>
```

Comment:
Rather than repeat the javadoc can we put it in the more specific method and reference it from the other?